boot/bootutil/src/bootutil_priv.h

```diff
@@ -166,6 +170,10 @@ struct boot_swap_state {
 
 _Static_assert(BOOT_IMAGE_NUMBER > 0, "Invalid value for BOOT_IMAGE_NUMBER");
 
+#if defined(MCUBOOT_XIP) && (BOOT_IMAGE_NUMBER != 1)
+#error "The MCUBOOT_XIP mode only supports single-image boot (MCUBOOT_IMAGE_NUMBER=1)."
```

Comment:
Do we plan on adding support for this at some point? I think the main constraint will be that you always have to use either both images from the first slot, or both images from the second slot.

Reply:
I think it's possible to support multi image boot:
- Secure image can pass the address of the veneer region when passing execution to non-secure. It can pass through the registers. Usually registers just cleared before calling NS, but nothing stops us to load there valuable info there.
- Then NS image can save this address to a global variable in Reset_Handler().
- Before a veneer call, the address of the actual valid veneer function could be calculated. Just calculate the offset of the veneer region( compile time fixed veneer region address - runtime received veneer region address) and add to the original static function address.
- I think this way always the active image veneer region would be called. 
- Also Secure code need the info that which is the active NS image to configure the memory isolation hw (MPC & SAU) properly. But this data can be propagated from bootloader to active secure image through the shared data area.

Reply:
Do we need this complexity, or just require the two images to always be considered together?

Reply:
If someone want to independently update the images (on arm-v8m S and NS) then I think we need this. 
Why might need independent update, I can think of:

- NS image usually more frequently updated then secure. Bandwidth constrains.
- Device operator not owning the secure image just the NS, only can update that one. Might different entity can update S image.

But I would be interested in from chip vendor perspective as well.

Reply:
This might be better solved by enhancing the update protocol to be able to be a delta off of the existing image.